src/wristpy/core/orchestrator.py

```diff
@@ -159,14 +99,132 @@ def run(
+        calibrator: The calibrator to be used on the input data.
+        epoch_length: The temporal resolution in seconds, the data will be down sampled
+            to. If None is given no down sampling is preformed.
+        verbosity: The logging level for the logger.
+        output_filetype: Specifies the data format for the save files. Must be None when
+            processing files, must be a valid file type when processing directories.
+
+    Returns:
+        All calculated data in a save ready format as a Results object or as a
+        dictionary of Results objects.
+
+    Raises:
+        ValueError: If processing a file and the output_filetype is not None
+        ValueError: If processing a directory and output_filetype is not a valid type.
+        ValueError: If processing a directory and the output given is not a directory.
+        FileNotFoundError: If the input directory contained no files of a valid type.
+
+
+    References:
+        [1] Hildebrand, M., et al. (2014). Age group comparability of raw accelerometer
+        output from wrist- and hip-worn monitors. Medicine and Science in Sports and
+        Exercise, 46(9), 1816-1824.
+    """
+    logger.setLevel(verbosity)
+
+    input = pathlib.Path(input)
+
+    if input.is_file():
+        if output_filetype is not None:
+            raise ValueError(
+                "When processing single files, output_filetype should be None - "
+                "the file type will be determined from the output path."
+            )
+        logger.debug("Input is file, forwarding to run_file with output=%s", output)
+
+        return run_file(
+            input=input,
+            output=output,
+            thresholds=thresholds,
+            calibrator=calibrator,
+            epoch_length=epoch_length,
+            verbosity=verbosity,
+        )
+    if output is not None:
+        output = pathlib.Path(output)
+        if output.is_file():
+            raise ValueError(
+                "Output is a file, but must be a directory when input is a directory."
+            )
+
+    if output_filetype is None or output_filetype not in VALID_FILE_TYPES:
+        raise ValueError(
+            "Invalid output_filetype: "
+            f"{output_filetype}. Valid options are: {VALID_FILE_TYPES}."
+        )
+
+    file_names = list(itertools.chain(input.glob("*.gt3x"), input.glob("*.bin")))
+
+    if not file_names:
+        raise FileNotFoundError(f"Directory {input} contains no .gt3x or .bin files.")
+
+    results_dict = {}
+    for file in file_names:
+        output_file_path = (
+            output / pathlib.Path(file.stem).with_suffix(output_filetype)
+            if output
+            else None
+        )
+        logger.debug(
+            "Processing directory: %s, current file: %s, save path: %s",
+            input,
+            file,
+            output_file_path,
+        )
+        try:
+            results_dict[file.stem] = run_file(
```

Comment:
Include the suffix in the key; this will overwrite when you have e.g. `subject1.bin` and `subject1.gt3x` in the same directory.